sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/SyncAccountRepository.kt

```diff
@@ -114,58 +118,72 @@ class AppSyncAccountRepository @Inject constructor(
         }
     }
 
-    override fun processCode(stringCode: String): Result<Boolean> {
-        val decodedCode: String? = kotlin.runCatching {
-            return@runCatching stringCode.decodeB64()
-        }.getOrNull()
-        if (decodedCode == null) {
-            Timber.w("Failed while b64 decoding barcode; barcode is unusable")
-            return Error(code = INVALID_CODE.code, reason = "Failed to decode code")
-        }
-
-        kotlin.runCatching {
-            Adapters.recoveryCodeAdapter.fromJson(decodedCode)?.recovery
-        }.getOrNull()?.let {
-            Timber.d("Sync: code is a recovery code")
-            return login(it)
-        }
+    override fun processCode(code: CodeType): Result<Boolean> {
+        when (code) {
+            is Recovery -> {
+                Timber.d("Sync: code is a recovery code")
+                return login(code.code)
+            }
 
-        kotlin.runCatching {
-            Adapters.recoveryCodeAdapter.fromJson(decodedCode)?.connect
-        }.getOrNull()?.let {
-            Timber.d("Sync: code is a connect code")
-            return connectDevice(it)
-        }
+            is Connect -> {
+                Timber.d("Sync: code is a connect code")
+                return connectDevice(code.code)
+            }
 
-        kotlin.runCatching {
-            Adapters.invitationCodeAdapter.fromJson(decodedCode)?.exchangeKey
-        }.getOrNull()?.let {
-            if (!syncFeature.exchangeKeysToSyncWithAnotherDevice().isEnabled()) {
-                Timber.w("Sync: Scanned exchange code type but exchanging keys to sync with another device is disabled")
-                return@let null
+            is Exchange -> {
+                if (!syncFeature.exchangeKeysToSyncWithAnotherDevice().isEnabled()) {
+                    Timber.w("Sync: Scanned exchange code type but exchanging keys to sync with another device is disabled")
+                } else {
+                    return onInvitationCodeReceived(code.code)
+                }
             }
 
-            return onInvitationCodeReceived(it)
+            else -> {
+                Timber.d("Sync: code type unknown")
+            }
         }
-
-        Timber.e("Sync: code is not supported")
+        Timber.e("Sync: code type (${code.javaClass.simpleName}) is not supported")
         return Error(code = INVALID_CODE.code, reason = "Failed to decode code")
     }
 
     override fun getCodeType(stringCode: String): CodeType {
```

Comment:
Now that this returns type + code in a data class, we might need to change the name to something different for clarity. GetCodeType made sense when having an enum and just returning the type.

Reply:
can call it `parseSyncAuthCode` which pairs better with the new return type `SyncAuthCode` and better reflects that it's not just getting the _type_